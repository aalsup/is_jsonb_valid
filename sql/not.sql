-- not
-- allowed
SELECT is_jsonb_valid('{"not":{"type":"integer"}}', '"foo"');
-- disallowed
SELECT is_jsonb_valid('{"not":{"type":"integer"}}', '1');
-- not multiple types
-- valid
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', '"foo"');
-- mismatch
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', '1');
-- other mismatch
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', 'true');
-- not more complex schema
-- match
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '1');
-- other match
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '{"foo":1}');
-- mismatch
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '{"foo":"bar"}');
-- forbidden property
-- property present
SELECT is_jsonb_valid('{"properties":{"foo":{"not":{}}}}', '{"foo":1,"bar":2}');
-- property absent
SELECT is_jsonb_valid('{"properties":{"foo":{"not":{}}}}', '{"bar":1,"baz":2}');