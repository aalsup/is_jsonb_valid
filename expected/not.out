-- not
-- allowed
SELECT is_jsonb_valid('{"not":{"type":"integer"}}', '"foo"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- disallowed
SELECT is_jsonb_valid('{"not":{"type":"integer"}}', '1');
 is_jsonb_valid 
----------------
 f
(1 row)

-- not multiple types
-- valid
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', '"foo"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- mismatch
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', '1');
 is_jsonb_valid 
----------------
 f
(1 row)

-- other mismatch
SELECT is_jsonb_valid('{"not":{"type":["integer","boolean"]}}', 'true');
 is_jsonb_valid 
----------------
 f
(1 row)

-- not more complex schema
-- match
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- other match
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '{"foo":1}');
 is_jsonb_valid 
----------------
 t
(1 row)

-- mismatch
SELECT is_jsonb_valid('{"not":{"type":"object","properties":{"foo":{"type":"string"}}}}', '{"foo":"bar"}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- forbidden property
-- property present
SELECT is_jsonb_valid('{"properties":{"foo":{"not":{}}}}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- property absent
SELECT is_jsonb_valid('{"properties":{"foo":{"not":{}}}}', '{"bar":1,"baz":2}');
 is_jsonb_valid 
----------------
 t
(1 row)

